dlt/pipeline/pipeline.py

```diff
@@ -1372,6 +1372,13 @@ def _maybe_destination_capabilities(
             if injected_caps:
                 injected_caps.__exit__(None, None, None)
 
+    def _normalize_dataset_name(self, dataset_name: str) -> str:
+        """Normalizes the dataset name using the dataset_name_layout."""
+        if "%s" not in self.config.dataset_name_layout:
+            raise ValueError("The dataset_name_layout must contain a '%s' placeholder.")
```

Comment:
You could tell the user why we need `%s` (to substitute `dataset_name` from the pipeline). Sometimes we even give examples so users can self-correct without reading more docs or coming to our slack